src/blocks/networkmanager.rs

```diff
@@ -28,125 +32,371 @@ enum NetworkState {
 impl From<u32> for NetworkState {
     fn from(id: u32) -> Self {
         match id {
-            // https://developer.gnome.org/NetworkManager/unstable/nm-dbus-types.html#NMState
+            // https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMState
             10 => NetworkState::Asleep,
             20 => NetworkState::Disconnected,
             30 => NetworkState::Disconnecting,
             40 => NetworkState::Connecting,
             50 => NetworkState::ConnectedLocal,
             60 => NetworkState::ConnectedSite,
             70 => NetworkState::ConnectedGlobal,
-            _  => NetworkState::Unknown,
+            _ => NetworkState::Unknown,
         }
     }
 }
 
-impl fmt::Display for NetworkState {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+enum ActiveConnectionState {
+    Unknown = 0,
+    Activating = 1,
+    Activated = 2,
+    Deactivating = 3,
+    Deactivated = 4,
+}
+
+impl From<u32> for ActiveConnectionState {
+    fn from(id: u32) -> Self {
+        match id {
+            // https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMActiveConnectionState
+            1 => ActiveConnectionState::Activating,
+            2 => ActiveConnectionState::Activated,
+            3 => ActiveConnectionState::Deactivating,
+            4 => ActiveConnectionState::Deactivated,
+            _ => ActiveConnectionState::Unknown,
+        }
+    }
+}
+
+impl ActiveConnectionState {
+    fn to_state(&self, good: &State) -> State {
         match self {
-            NetworkState::Unknown => write!(f, "DOWN"),
-            NetworkState::Asleep => write!(f, "DOWN"),
-            NetworkState::Disconnected => write!(f, "DOWN"),
-            NetworkState::Disconnecting => write!(f, "DOWN"),
-            NetworkState::Connecting => write!(f, "DOWN"),
-            NetworkState::ConnectedLocal => write!(f, "UP"),
-            NetworkState::ConnectedSite => write!(f, "UP"),
-            NetworkState::ConnectedGlobal => write!(f, "UP"),
+            ActiveConnectionState::Activated => good.clone(),
+            ActiveConnectionState::Activating => State::Warning,
+            ActiveConnectionState::Deactivating => State::Warning,
+            ActiveConnectionState::Deactivated => State::Critical,
+            ActiveConnectionState::Unknown => State::Critical,
         }
     }
 }
 
-enum ConnectionType {
-    Ethernet,
-    Wireless,
-    Other,
+#[derive(Debug)]
+enum DeviceType {
```

Comment:
So I understand why you've included all possible devices, but many of them will not be encountered by usters. Can you simplify the code by amalgamating the more unusual devices into an `Other` discriminant?

Reply:
Of course. The documentation link should be sufficient to extend as necessary.

I believe I primarily added them all for exploratory reasons.